satpy/utils.py

```diff
@@ -574,29 +574,59 @@ def ignore_invalid_float_warnings():
         yield
 
 
-def get_chunk_size_limit(dtype):
-    """Compute the chunk size limit in bytes given *dtype*.
+def get_chunk_size_limit(dtype=float):
+    """Compute the chunk size limit in bytes given *dtype* (float by default).
+
+    It is derived from PYTROLL_CHUNK_SIZE if defined (although deprecated) first, from dask config's `array.chunk-size`
+    then. It defaults to 128MiB.
 
     Returns:
-        If PYTROLL_CHUNK_SIZE is not defined, this function returns None,
-        otherwise it returns the computed chunk size in bytes.
+        The recommended chunk size in bytes.
     """
-    pixel_size = get_chunk_pixel_size()
+    pixel_size = _get_chunk_pixel_size()
     if pixel_size is not None:
         return pixel_size * np.dtype(dtype).itemsize
-    return None
+    dask_chunk_size = dask.config.get("array.chunk-size", "128MiB")
+    return dask.utils.parse_bytes(dask_chunk_size)
```

Comment:
Something about this logic being in two places in this module smells funny to me. If you think the code is cleaner with it this way I am OK with that. I just wanted to point it out.

Reply:
Fixed, thanks